commit e185b99c87a899af32d5b6ac5a75e0655102926b (refs/changes/65/865/1)
Author: Richard Bradley <richard.bradley@joyent.com>
Date:   2016-11-09T16:53:41+00:00 (2 years, 11 months ago)
    
    MANTA-2587 manta-net.sh Needs to Support Adding tags to Aggregations

diff --git a/docs/operator-guide/index.md b/docs/operator-guide/index.md
index 37270fd..7fdfc00 100644
--- a/docs/operator-guide/index.md
+++ b/docs/operator-guide/index.md
@@ -833,16 +833,16 @@ for details on how to check if services are healthy.
 The networking configuration file is a per-datacenter JSON file with several
 properties:
 
-| Property       | Kind                       | Description                                                                                                                                                                                   |
-| -------------- | -------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| `azs`          | array&nbsp;of&nbsp;strings | list of all availabililty zones (datacenters) participating in Manta in this region                                                                                                           |
-| `this_az`      | string                     | string (in `azs`) denoting this availability zone                                                                                                                                             |
-| `manta_nodes`  | array&nbsp;of&nbsp;strings | list of server uuid's for *all* servers participating in Manta in this AZ                                                                                                                     |
-| `marlin_nodes` | array&nbsp;of&nbsp;strings | list of server uuid's (subset of `manta_nodes`) that are storage nodes                                                                                                                        |
-| `admin`        | object                     | describes the "admin" network in this datacenter (see below)                                                                                                                                  |
-| `manta`        | object                     | describes the "manta" network in this datacenter (see below)                                                                                                                                  |
-| `marlin`       | object                     | describes the "marlin" network in this datacenter (see below)                                                                                                                                 |
-| `mac_mappings` | object                     | maps each server uuid from `manta_nodes` to an object mapping each network name ("admin", "manta", and "marlin") to the MAC address on that server over which that network should be created. |
+| Property       | Kind                       | Description                                                                                                                                                                          |
+| -------------- | -------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `azs`          | array&nbsp;of&nbsp;strings | list of all availabililty zones (datacenters) participating in Manta in this region                                                                                                  |
+| `this_az`      | string                     | string (in `azs`) denoting this availability zone                                                                                                                                    |
+| `manta_nodes`  | array&nbsp;of&nbsp;strings | list of server uuid's for *all* servers participating in Manta in this AZ                                                                                                            |
+| `marlin_nodes` | array&nbsp;of&nbsp;strings | list of server uuid's (subset of `manta_nodes`) that are storage nodes                                                                                                               |
+| `admin`        | object                     | describes the "admin" network in this datacenter (see below)                                                                                                                         |
+| `manta`        | object                     | describes the "manta" network in this datacenter (see below)                                                                                                                         |
+| `marlin`       | object                     | describes the "marlin" network in this datacenter (see below)                                                                                                                        |
+| `nic_mappings` | object                     | maps each server in both `manta_nodes` and `marlin_nodes` to an object mapping each network name ("manta" and "marlin") to the network interface on the server that should be tagged |
 
 "admin", "manta", and "marlin" all describe these networks that are built into Manta:
 
@@ -865,6 +865,23 @@ Besides those two, each of these blocks has a property for the current
 availability zone that describes the "subnet", "gateway", "vlan_id", and
 "start" and "end" provisionable addresses.
 
+`nic_mappings` contains a list of servers representing their intended
+NIC tag configuration. The following properties are expected in this list.
+
+| Property            | Kind   | Description                                                                                                           |
+| ------------------- | ------ | --------------------------------------------------------------------------------------------------------------------- |
+| `$server_uuid`      | object | each property of this object should be the NIC tag of the network to be added                                         |
+| `$server_uuid.$tag` | object | an object whose property is either "mac" or "aggr", and value is the MAC address or the Aggregation name to be tagged |
+
+`mac_mappings` is also supported in place of `nic_mappings` to provide
+some form of backwards compatibility. The only difference is that
+`$server_uuid.$tag` should contain a string representing the MAC address
+of the server to be tagged. Only one of `nic_mappings` or `mac_mappings`
+is supported per network configuration file.
+
+**Note:** Aggregations must already exist in NAPI, and updating NIC tags on
+aggregations will require a reboot of the server in question.
+
 For reference, here's an example multi-datacenter configuration with one service
 node (aac3c402-3047-11e3-b451-002590c57864) and one storage node
 (445aab6c-3048-11e3-9816-002590c3f3bc):
@@ -953,17 +970,39 @@ node (aac3c402-3047-11e3-b451-002590c57864) and one storage node
           "gateway": "172.28.128.1"
         }
       },
+    },
 
-      "mac_mappings": {
-        "aac3c402-3047-11e3-b451-002590c57864": {
-          "manta": "90:e2:ba:4b:ec:d1"
+    "nic_mappings": {
+      "aac3c402-3047-11e3-b451-002590c57864": {
+        "manta": {
+          "mac": "90:e2:ba:4b:ec:d1"
+        }
+      },
+      "445aab6c-3048-11e3-9816-002590c3f3bc": {
+        "manta": {
+          "mac": "90:e2:ba:4a:32:71"
         },
-        "445aab6c-3048-11e3-9816-002590c3f3bc": {
-          "manta": "90:e2:ba:4a:32:71",
-          "mantanat": "90:e2:ba:4a:32:71"
+        "mantanat": {
+          "aggr": "aggr0"
         }
       }
     }
+ 
+The following is an example of the `mac_mappings` structure that only 
+supports tagging MAC addresses.
+
+    ...
+    },
+
+    "mac_mappings": {
+      "aac3c402-3047-11e3-b451-002590c57864": {
+        "manta": "90:e2:ba:4b:ec:d1"
+      },
+      "445aab6c-3048-11e3-9816-002590c3f3bc": {
+        "manta": "90:e2:ba:4a:32:71",
+        "mantanat": "90:e2:ba:4a:32:71"
+      }
+    }
 
 In a multi-datacenter configuration, this would be used to configure the
 "staging-1" datacenter.  There would be two more configuration files, one
